.github/workflows/publish.yml

```diff
@@ -0,0 +1,29 @@
+name: PyPI Automated publish
+
+on:
+  release:
+    types: [created]
```

Comment:
Can we also run this workflow for pull requests? I think it would be nice if we test the publish by running `poetry publish --dry-run` when a pull request is created. What do you think?

Reply:
That makes a lot of sense, I wasn't aware there was such a flag for poetry. I need to read more docs!

Reply:
Added changes in latest commit.